mos_tests/functions/os_cli.py

```diff
@@ -65,19 +65,36 @@ def __call__(self, action, flags='', params='', prefix='', fail_ok=False,
 class OpenStack(CLICLient):
     command = 'openstack'
 
-    def details(self, output):
+    def details(self, output, mapping=('Field', 'Value')):
+        """List with one dict with data"""
         data = json.loads(output)
         if isinstance(data, list):
-            data = {x['Field']: x['Value'] for x in data}
+            data = {x[mapping[0]]: x[mapping[1]] for x in data}
         return data
 
+    def listing(self, output, mapping=False):
+        """List with several dicts with data"""
+        data = json.loads(output)
+        if mapping:
+            if isinstance(data, list):
+                data = [{x[mapping[0]]: x[mapping[1]]} for x in data]
+        return data
+
+    def project_list(self):
+        output = self('project list -f json')
+        return self.listing(output, ('Name', 'ID'))
```

Comment:
Method name says, that this method returns list (or tuple, or generator) object.